ca/ssl.go

```
@@ -0,0 +1,143 @@
+			Country:       []string{"COUNTRY_CODE"},
+			Province:      []string{"PROVINCE"},
+			Locality:      []string{"CITY"},
+			StreetAddress: []string{"ADDRESS"},
+			PostalCode:    []string{"POSTAL_CODE"},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().AddDate(10, 0, 0),
+		IsCA:                  true,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		BasicConstraintsValid: true,
+	}
+
+	pub := &priv.PublicKey
+	ca_b, err := x509.CreateCertificate(rand.Reader, ca, ca, pub, priv)
+	if err != nil {
+		log.Println("create ca failed", err)
+		return nil, nil, nil, err
+	}
+
+	// Public key
+	certOut, err := os.Create("ca.crt")
+	pem.Encode(certOut, &pem.Block{Type: "CERTIFICATE", Bytes: ca_b})
+	certOut.Close()
+	log.Print("written cert.pem\n")
+
+	// Private key
+	keyOut, err := os.OpenFile("ca.key", os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	pem.Encode(keyOut, &pem.Block{Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(priv)})
+	keyOut.Close()
+	log.Print("written key.pem\n")
+
+	return ca, ca_b, priv, nil
+}
+
+func serverCert(caPrivKey *rsa.PrivateKey, ca *x509.Certificate) (*rsa.PrivateKey, []byte, error) {
+
+	cert := &x509.Certificate{
+		SerialNumber: big.NewInt(2019),
+		Subject:      baseName,
+		IPAddresses:  []net.IP{net.IPv4(127, 0, 0, 1), net.IPv6loopback},
+		NotBefore:    time.Now(),
+		NotAfter:     time.Now().AddDate(10, 0, 0),
+		SubjectKeyId: []byte{1, 2, 3, 4, 6},
+		ExtKeyUsage:  []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage:     x509.KeyUsageDigitalSignature,
+		DNSNames:     []string{"localhost", "example.com", "127.0.0.1"},
+	}
+
+	pk, _ := getRSA()
+	cert_b, _ := x509.CreateCertificate(rand.Reader, cert, ca, &pk.PublicKey, caPrivKey)
+
+	// Public key
+	certOut, _ := os.Create("server.crt")
```

Why are we ignoring errors here?

---

its just a script so I didnt even much time on as on the challenge doc it was mentioned you have your own set of certificates. Ill make it more resilient